### DISCUSSION TITLE
src.csharp hidden in VS Code

### DISCUSSION THREAD

**Discussion by jonaro00** [CoderLine/alphaTab#1034] [Q&A] [answered]
Some time ago when I was fixing C# test cases I noticed that the src.csharp folder wasn't showing in VS Code. Today I found why:

In `.vscode/settings.json`:
```json
    "files.exclude": {
        "dist/lib.csharp/**": true,
        "src.csharp/**": true,
        "node_modules/**": true
    }
```

Is there a reason for this? Perhaps hiding some subfolder with gitignored binaries is the intention?

**Comment by jonaro00** [comment]
I can also add a question: Why is `src/generated` not gitignored? Any specific use case in environments with no capability to build?

**Comment by Danielku15** [comment, marked as answer]
The C# code is ignored on the top level project mainly because when working on the C# code I have a separate Visual Studio / Rider open and do not want to have the C# code spoil the code navigation within TypeScript. Having it not ignored, causes it to be included in all searches and code completion actions which most of the time is not the desired location to jump to. Actually my ultimate goal is that there is no need for having C# code but have all code in TypeScript. But I never went that far eliminating the C#/Kotlin code until now. 

The discussion about `src/generated` has been started already multiple times. The main reason is despite the nature of it being generated code, it makes any changes done on the model clearly visible also in the serializer. If there is a change in the model, but no matching change in the serializer code this rings alarm bells and avoids bugs.
